invenio_rdm_records/collections/schema.py

```diff
@@ -14,7 +14,8 @@ class CollectionSchema(Schema):
 
     slug = fields.Str()
     title = fields.Str()
-    depth = fields.Int()
+    depth = fields.Int(dump_only=True)
```

Comment:
won't this prevent us from creating collections with desired depth via service in the future ?

Reply:
No, the `depth` of a collection tells you how deep it is in the tree and is a `read-only` field. It is computed based on the `path` when you create a collection. The `path` is the field that matters here since collections are based on the [materialized path pattern](https://www.mongodb.com/docs/manual/tutorial/model-tree-structures-with-materialized-paths/). 

Initially, we added `depth` as a field to improve the performance of read queries.

The service has two methods to create collections:

- `create`: creates a new "root" collection (e.g. no parent) inside a `CollectionTree`. The path is always "empty" since it's a root (`path=','`).  I added a docstring to be more explicit about it. 
- `add`: adds a collection as a child of another one. The path will be computed based on the parent.